### PR TITLE
Refactor slicing test runner and get rid of UB in tests

### PR DESCRIPTION
**Pull request by lzaoral**
Some small fixes and improvements that I've additionally made during #412 but are independent of it.

### CODE REVIEW

File: tests/slicing/sources/bitcast2.c [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
     char *byte = (char *) &a;
-    byte[0] = 0xab;
-    byte[1] = 0xab;
-    byte[2] = 0xab;
-    byte[3] = 0xab;
-    test_assert(a == 0xabababab);
+
+    int result = 0;
+    for (unsigned i = 0; i < sizeof(int); i++)
+        result |= (byte[i] = 0x1b) << i * CHAR_BIT;
```
Comment (mchalupa) [inline tests/slicing/sources/bitcast2.c:10]: This completely changes the test, at least from the point of view of the slicer. Accessing the bytes via constants (e.g., `byte[1]`) leads to a precise knowledge of what memory is accessed, while accessing it via a variable (`bytes[i]`) leads to over-approximating the precise knowledge. Please, keep the spirit of the test as it was (accessing bytes via constants). If you want this test with the loop, just add it as a new test (but check that there is not a similar test already).

**Review by mchalupa — CHANGES_REQUESTED** [review]

File: tests/slicing/sources/bitcast2.c (hunk 1 above)
Comment (lzaoral) [reply tests/slicing/sources/bitcast2.c:10]: Oh, I see. I wanted to make the test platform as platform independent as possible so didn't even think about this. I'll revert this change and add this test as a new one. There is already a rather similar one but that one is accessing the bytes by `*byte; byte++` pattern and not by `byte[i]` so I guess that's ok.

Comment (lzaoral) [reply tests/slicing/sources/bitcast2.c]: Nevertheless, I'll change the the type of the `char` pointer to be explicitly `unsigned` to silence a `clang-tidy` warning about implementation defined behaviour of a narrowing conversion from `int` to `char` for values greater than `127` which `0xab` is.

**Comment by mchalupa** [comment]
Very well, let's merge this then. Thanks a lot!

**Pull request merged**